src/test/basics/PerfLog_test.cpp

```diff
@@ -0,0 +1,1075 @@
+            BEAST_EXPECT(currentJson[jss::methods].size() == 0);
+        };
+
+        // Start an ID that's too large.
+        perfLog->jobStart (jobType, microseconds {11}, steady_clock::now(), 2);
+        std::this_thread::sleep_for (microseconds {10});
+        verifyCounters (perfLog->countersJson(), 1, 0, 11, 0);
+        verifyEmptyCurrent (perfLog->currentJson());
+
+        // Start a negative ID
+        perfLog->jobStart (jobType, microseconds {13}, steady_clock::now(), -1);
+        std::this_thread::sleep_for (microseconds {10});
+        verifyCounters (perfLog->countersJson(), 2, 0, 24, 0);
+        verifyEmptyCurrent (perfLog->currentJson());
+
+        // Finish the too large ID
+        perfLog->jobFinish (jobType, microseconds {17}, 2);
+        std::this_thread::sleep_for (microseconds {10});
+        verifyCounters (perfLog->countersJson(), 2, 1, 24, 17);
+        verifyEmptyCurrent (perfLog->currentJson());
+
+        // Finish the negative ID
+        perfLog->jobFinish (jobType, microseconds {19}, -1);
+        std::this_thread::sleep_for (microseconds {10});
+        verifyCounters (perfLog->countersJson(), 2, 2, 24, 36);
+        verifyEmptyCurrent (perfLog->currentJson());
+
+        // Give the PerfLog enough time to flush it's state to the file.
+        std::this_thread::sleep_for (parent.getLogInterval() * 3);
+
+        // Politely stop the PerfLog.
+        parent.doStop();
+
+        // Check file contents if that is appropriate.
+        auto const fullPath =
+            parent.getPerfLogPath() / parent.getPerfLogFileName();
+
+        if (withFile == WithFile::no)
+        {
+            BEAST_EXPECT(! exists (fullPath));
+        }
+        else
+        {
+            // The last line in the log file should contain the same
+            // information that countersJson() and currentJson() returned.
+            // Verify that.
+
+            // Get the last line of the log.
+            std::ifstream log (fullPath.c_str());
+            std::string lastLine;
+            for (std::string line; std::getline (log, line); )
+            {
+                if (! line.empty())
+                    lastLine = std::move (line);
+            }
+
+            Json::Value parsedLastLine;
+            Json::Reader ().parse (lastLine, parsedLastLine);
+            if (! BEAST_EXPECT(! RPC::contains_error (parsedLastLine)))
+                // Avoid cascade of failures
+                return;
+
+            // Validate the contents of the last line of the log.
+            verifyCounters (parsedLastLine[jss::counters], 2, 2, 24, 36);
+            verifyEmptyCurrent (parsedLastLine[jss::current_activities]);
+        }
+    }
+
+    void testRotate (WithFile withFile)
+    {
+        // We can't fully test rotate because unit tests must run on Windows,
+        // and Windows doesn't (may not?) support rotate.  But at least call
+        // the interface and see that it doesn't crash.
```

Comment:
I'd be okay with some additional tests/assertions being in an `ifndef` for windows - that would give us some a little extra coverage on one/two other platforms.